internal/storage/bucket_handle.go

```diff
@@ -72,6 +72,32 @@ func (b *bucketHandle) DeleteObject(ctx context.Context, req *gcs.DeleteObjectRe
 		obj = obj.If(storage.Conditions{MetagenerationMatch: *req.MetaGenerationPrecondition})
 	}
 
-	// Deleting object through Go Storage Client.
 	return obj.Delete(ctx)
 }
+
+func (b *bucketHandle) StatObject(ctx context.Context, req *gcs.StatObjectRequest) (o *gcs.Object, err error) {
+	var attrs *storage.ObjectAttrs
+	// Retrieving object attrs through Go Storage Client.
+	attrs, err = b.bucket.Object(req.Name).Attrs(ctx)
+
+	// If error is of type storage.ErrObjectNotExist, then we have to retry once by appending '/' to the object name.
+	// We are retyring to handle the case when the object is a directory.
+	// Since directories in GCS bucket are denoted with a an extra '/' at the end of their name. But in the request we are only provided with their name without '/'.
+	if err == storage.ErrObjectNotExist {
```

Comment:
We are replicating the jacobsa/bucket.go behaviour here. https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/vendor/github.com/jacobsa/gcloud/gcs/bucket.go#L213
I dont see this extra check in jacobsa's code. Also LookupInode code takes care of handling these scenarios: https://github.com/GoogleCloudPlatform/gcsfuse/blob/fa9807bd3d424e8f398c3eb059038fe772052bc4/internal/fs/fs.go#L727.  Can you remove this condition and try to invoke LookupInode calls for file, directory, implicit dir and see if it works.

Reply:
fixed